packages/core/src/browser/keybinding.ts

```diff
@@ -188,12 +189,13 @@ export class KeybindingRegistry {
     }
 
     /**
-     * Register a default keybinding to the registry.
+     * Register a keybinding to the registry.
      *
      * @param binding
+     * @param scope keybinding scope, {@link KeybindingScope.DEFAULT} if not provided.
      */
-    registerKeybinding(binding: Keybinding): Disposable {
-        return this.doRegisterKeybinding(binding, KeybindingScope.DEFAULT);
+    registerKeybinding(binding: Keybinding, scope?: KeybindingScope): Disposable {
```

Comment:
We should not allow random scopes, but rather have a boolean flag, i.e. something like `overriding: boolean`.
Keybindings for other scopes should be registered with `setKeymaps`.

Reply:
done